src/com/tictacgo/DirectionPickerFragment.java

```diff
@@ -0,0 +1,141 @@
+  }
+
+  /**
+   * Creates and returns a new DirectionPicker object.
+   *
+   * @param player The player whose turn it is.
+   * @param gravity The gravity of the location that was just pressed.
+   * @param height The height of the board.
+   * @return A new DirectionPicker object with the given parameters.
+   */
+  public static DirectionPickerFragment getInstance(Player player, int gravity, int height) {
+    Bundle arguments = new Bundle();
+    arguments.putSerializable(playerArgumentKey, player);
+    arguments.putInt(gravityArgumentKey, gravity);
+    arguments.putInt(heightArgumentKey, height);
+
+    DirectionPickerFragment directionPicker = new DirectionPickerFragment();
+    directionPicker.setArguments(arguments);
+    return directionPicker;
+  }
+
+  @Override
+  public void setArguments(Bundle arguments) {
+    player = (Player) arguments.getSerializable(playerArgumentKey);
+    height = arguments.getInt(heightArgumentKey);
+    gravity = arguments.getInt(gravityArgumentKey);
+  }
+
+  @Override
+  public void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    listener = (OnDirectionPickedListener) getActivity();
+
+    directionClicked = new View.OnClickListener() {
+      public void onClick(View v) {
+       switch (v.getId()) { //Which direction was picked
+          case R.id.directionTopLeft:
+            listener.onDirectionPicked(-1, -1, gravity);
+            break;
+          case R.id.directionTopMiddle:
+            listener.onDirectionPicked(-1, 0, gravity);
+            break;
+          case R.id.directionTopRight:
+            listener.onDirectionPicked(-1, 1, gravity);
+            break;
+          case R.id.directionMiddleLeft:
+            listener.onDirectionPicked(0, -1, gravity);
+            break;
+          case R.id.directionMiddleRight:
+            listener.onDirectionPicked(0, 1, gravity);
+            break;
+          case R.id.directionBottomLeft:
+            listener.onDirectionPicked(1, -1, gravity);
+            break;
+          case R.id.directionBottomMiddle:
+            listener.onDirectionPicked(1, 0, gravity);
+            break;
+          case R.id.directionBottomRight:
+            listener.onDirectionPicked(1, 1, gravity);
+            break;
+          default:
+            getFragmentManager().popBackStack();
+        }
+      }
+    };
+  }
+
+  @Override
+  public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                           Bundle savedInstanceState) {
+    // Create and locate view properly
+    View view = inflater.inflate(R.layout.fragment_direction_picker, container, false);
+    LayoutParams params = new LayoutParams(LayoutParams.WRAP_CONTENT, LayoutParams.WRAP_CONTENT,
+        gravity);
+    view.setLayoutParams(params);
+
+    int id = player == Player.X ? R.drawable.piecexdirection : R.drawable.pieceodirection;
+
+    // Set direction buttons
```

Comment:
We repeat the same numbers here and in Piece, and they're not super readable -- it would be nice to eventually pull them out into an enum or a bunch of constants in an Angles class or something.